x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts

```diff
@@ -438,18 +435,24 @@ export const relatedEventInfoByEntityId: (
 );
 
 /**
- * If we need to fetch, this is the ID to fetch.
+ * If the tree resource needs to be fetched then these are the parameters that should be used.
  */
-export function databaseDocumentIDToFetch(state: DataState): string | null {
-  // If there is an ID, it must match either the last received version, or the pending version.
-  // Otherwise, we need to fetch it
-  // NB: this technique will not allow for refreshing of data.
+export function treeParametersToFetch(state: DataState): TreeFetcherParameters | null {
```

Comment:
The logic to determine if we need to make a request now accounts for changes in either `databaseDocumentID` or `indices`.